ksmt-symfpu/src/main/kotlin/org/ksmt/symfpu/solver/SymFPUModel.kt

```diff
@@ -0,0 +1,204 @@
+package org.ksmt.symfpu.solver
+
+import org.ksmt.KContext
+import org.ksmt.decl.KDecl
+import org.ksmt.expr.KApp
+import org.ksmt.expr.KArrayConst
+import org.ksmt.expr.KArrayLambdaBase
+import org.ksmt.expr.KArrayStoreBase
+import org.ksmt.expr.KConst
+import org.ksmt.expr.KExpr
+import org.ksmt.expr.KFunctionAsArray
+import org.ksmt.expr.KUninterpretedSortValue
+import org.ksmt.expr.transformer.KTransformer
+import org.ksmt.solver.KModel
+import org.ksmt.solver.model.KModelEvaluator
+import org.ksmt.solver.model.KModelImpl
+import org.ksmt.sort.KArraySortBase
+import org.ksmt.sort.KFpSort
+import org.ksmt.sort.KSort
+import org.ksmt.sort.KUninterpretedSort
+import org.ksmt.symfpu.operations.pack
+import org.ksmt.symfpu.solver.ArraysTransform.Companion.mkAnyArrayLambda
+import org.ksmt.symfpu.solver.ArraysTransform.Companion.mkAnyArrayStore
+import org.ksmt.utils.cast
+import org.ksmt.utils.uncheckedCast
+
+class SymFPUModel(private val kModel: KModel, val ctx: KContext, val transformer: FpToBvTransformer) : KModel {
+    private val mapBvToFpDecls
+        get() = transformer.mapFpToBvDecl.entries.associateBy({ it.value.decl }) { it.key }
+
+    override val declarations: Set<KDecl<*>>
+        get() = kModel.declarations.map { mapBvToFpDecls[it] ?: it }.toSet()
```

Comment:
`declarations.mapTo(hashSetOf()) { mapBvToFpDecls[it] ?: it }` to avoid intermediate list creation.